dvc/repo/experiments/__init__.py

```diff
@@ -852,12 +857,20 @@ def _get_baseline(self, rev):
     def _get_branch_containing(self, rev):
         from git.exc import GitCommandError
 
-        if self.scm.repo.head.is_detached:
-            self._checkout_default_branch()
```

Comment:
Faster to just avoid this checkout and drop the detached head branch entry ourselves.

Also, doing this checkout could introduce potential bugs where the caller was not expecting the checked out git commit to change.